### ISSUE TITLE
AI Error : infinite loop

### ISSUE THREAD

**Issue by jinlan** [freeorion/freeorion#2520]
```
[03 八月 09:45:10] AI_5: AI_Error: AI script error : FreeOrionAI.py:generateOrders():338  - Exception maximum recursion depth exceeded while calling a Python object while trying to issue_fleet_orders_for_fleet_missions
Traceback (most recent call last):
  File "F:/Game\fo\FreeOrion\default\python\AI\FreeOrionAI.py", line 335, in generateOrders
    action()
  File "F:/Game\fo\FreeOrion\default\python\AI\FleetUtilsAI.py", line 588, in issue_fleet_orders_for_fleet_missions
    mission.issue_fleet_orders()
  File "F:/Game\fo\FreeOrion\default\python\AI\AIFleetMission.py", line 509, in issue_fleet_orders
    allocations=allocations)
  File "F:/Game\fo\FreeOrion\default\python\AI\MilitaryAI.py", line 910, in assign_military_fleets_to_systems
    fleet_mission.generate_fleet_orders()
  File "F:/Game\fo\FreeOrion\default\python\AI\AIFleetMission.py", line 664, in generate_fleet_orders
    else TargetSystem(self._get_target_for_protection_mission()))
  File "F:/Game\fo\FreeOrion\default\python\AI\AIFleetMission.py", line 777, in _get_target_for_protection_mission
    self.issue_fleet_orders()
....
  File "F:/Game\fo\FreeOrion\default\python\AI\AIFleetMission.py", line 509, in issue_fleet_orders
    allocations=allocations)
  File "F:/Game\fo\FreeOrion\default\python\AI\MilitaryAI.py", line 910, in assign_military_fleets_to_systems
    fleet_mission.generate_fleet_orders()
  File "F:/Game\fo\FreeOrion\default\python\AI\AIFleetMission.py", line 664, in generate_fleet_orders
    else TargetSystem(self._get_target_for_protection_mission()))
  File "F:/Game\fo\FreeOrion\default\python\AI\AIFleetMission.py", line 777, in _get_target_for_protection_mission
    self.issue_fleet_orders()
  File "F:/Game\fo\FreeOrion\default\python\AI\AIFleetMission.py", line 509, in issue_fleet_orders
    allocations=allocations)
  File "F:/Game\fo\FreeOrion\default\python\AI\MilitaryAI.py", line 910, in assign_military_fleets_to_systems
    fleet_mission.generate_fleet_orders()
  File "F:/Game\fo\FreeOrion\default\python\AI\AIFleetMission.py", line 664, in generate_fleet_orders
    else TargetSystem(self._get_target_for_protection_mission()))
  File "F:/Game\fo\FreeOrion\default\python\AI\AIFleetMission.py", line 737, in _get_target_for_protection_mission
    debug("Trying to find target for protection mission. Target: %s", self.target)
  File "F:\Game\fo\FreeOrion\python27.zip\logging\__init__.py", line 1637, in debug
    root.debug(msg, *args, **kwargs)
  File "F:\Game\fo\FreeOrion\python27.zip\logging\__init__.py", line 1155, in debug
    self._log(DEBUG, msg, args, **kwargs)
  File "F:\Game\fo\FreeOrion\python27.zip\logging\__init__.py", line 1286, in _log
    self.handle(record)
  File "F:\Game\fo\FreeOrion\python27.zip\logging\__init__.py", line 1296, in handle
    self.callHandlers(record)
  File "F:\Game\fo\FreeOrion\python27.zip\logging\__init__.py", line 1336, in callHandlers
    hdlr.handle(record)
  File "F:\Game\fo\FreeOrion\python27.zip\logging\__init__.py", line 759, in handle
    self.emit(record)
  File "F:/Game/fo/FreeOrion/default/python\common\configure_logging.py", line 164, in emit
    record.message = record.getMessage()
  File "F:\Game\fo\FreeOrion\python27.zip\logging\__init__.py", line 329, in getMessage
    msg = msg % self.args
  File "F:/Game\fo\FreeOrion\default\python\AI\target.py", line 26, in __str__
    return str(target)
  File "F:/Game\fo\FreeOrion\default\python\AI\freeorion_tools\extend_freeorion_AI_interface.py", line 77, in system_to_string
    return to_str(SYSTEM, system.systemID, system.name)
RuntimeError: maximum recursion depth exceeded while calling a Python object
```

Comes out to message window while everything on my side is fine. Un-reproducable.
My version : build-2019-7-30

**Comment by Cjkjvfnby** [comment]
I have replaced multiple repeatable parts with \...\ in the original message.

**Comment by Morlic-fo** [comment]
Could you upload a savegame in xml format where the issue occurs so I can test and verify a fix?

**Comment by jinlan** [comment]
Started another game, auto-save is lost.

**Comment by jinlan** [comment]
Got that infinite loop again, 
This is my [save ](https://github.com/jinlan/issue/blob/master/FreeOrion_Ego_Ego_0130_20190808_170848.7z)

**Comment by Morlic-fo** [comment]
Thank you very much for that save. Should be fixed by #2527.
